src/main/kotlin/uk/gov/justice/digital/hmpps/hmppschallengesupportinterventionplanapi/entity/Referral.kt

```diff
@@ -88,6 +89,89 @@ data class Referral(
     referencedColumnName = "reference_Data_id",
   ) val incidentInvolvement: ReferenceData,
 ) : AbstractAggregateRoot<Referral>() {
+  @OneToOne(
+    mappedBy = "referral",
+    fetch = FetchType.LAZY,
+    cascade = [CascadeType.PERSIST, CascadeType.MERGE, CascadeType.REMOVE],
+  )
+  private var decisionAndActions: DecisionAndActions? = null
+
+  fun decisionAndActions() = decisionAndActions
+
+  fun createDecisionAndActions(
+    decisionOutcome: ReferenceData,
+    decisionOutcomeSignedOffBy: ReferenceData?,
+    decisionConclusion: String?,
+    decisionOutcomeRecordedBy: String,
+    decisionOutcomeRecordedByDisplayName: String,
+    decisionOutcomeDate: LocalDate?,
+    nextSteps: String?,
+    actionOther: String?,
+    actionedAt: LocalDateTime,
+    source: Source,
+    activeCaseLoadId: String?,
+    reason: Reason = Reason.USER,
+    actionOpenCsipAlert: Boolean,
+    actionNonAssociationsUpdated: Boolean,
+    actionObservationBook: Boolean,
+    actionUnitOrCellMove: Boolean,
+    actionCsraOrRsraReview: Boolean,
+    actionServiceReferral: Boolean,
+    actionSimReferral: Boolean,
+    description: String = "Decision and actions added to referral",
+  ): CsipRecord {
+    if (decisionAndActions != null) {
+      throw DecisionAndActionsAlreadyExistException(csipRecord.recordUuid)
+    }
+
+    decisionAndActions = DecisionAndActions(
+      referral = this,
+      decisionOutcome,
+      decisionOutcomeSignedOffBy,
+      decisionConclusion,
+      decisionOutcomeRecordedBy,
+      decisionOutcomeRecordedByDisplayName,
+      decisionOutcomeDate,
+      nextSteps,
+      actionOpenCsipAlert,
+      actionNonAssociationsUpdated,
+      actionObservationBook,
+      actionUnitOrCellMove,
+      actionCsraOrRsraReview,
+      actionServiceReferral,
+      actionSimReferral,
+      actionOther,
+    )
+
+    with(csipRecord) {
+      addAuditEvent(
+        action = AuditEventAction.UPDATED,
```

Comment:
the AuditEventAction is expected to be CREATED according to the ticket.

@iterative-it i assume that's not a copy and paste error? you do mean that audit event would record it as a `CREATED` action relating to the DecisionAndAction,
while in terms of domain event it's a `CsipUpdatedEvent`?

Reply:
No you're right - I'd mixed the two up. Now fixed.